src/Storage/Notmuch.hs

```diff
@@ -159,8 +160,10 @@ getFromNotmuchConfig :: String -> IO String
 getFromNotmuchConfig key = do
   let cmd = "notmuch"
   let args = ["config", "get", key]
-  stdout <- readProcess cmd args []
-  pure $ filter (/= '\n') stdout
+  (stdout, _) <- readProcess_ $ proc cmd args
```

Comment:
non-total (see http://hackage.haskell.org/package/typed-process-0.2.3.0/docs/System-Process-Typed.html#v:readProcess_).

We need to check exit code and handle the failure case (return `Nothing`?)

Reply:
Yes true. I changed it with the equivalent function, since the former function was non-total as well. Will change.

Reply:
Hm... so I looked into this and I think it's not worth changing now, because it's a bit of a rabbit hole.

The type for this configuration bit is `IO FilePath` which is converted to a `FilePath` when we process the config. What is missing at this stage is some kind of error handling. Since the function is non-total at the moment and used in the config only, it would basically enforce an error state if the config or notmuch is not correctly setup. Furthermore, it'll show the stderr message.

What happens if I change this to a `IO (Maybe FilePath)` is the fact that I'm deferring the evaluation of the error until pre-processing of the config happens. I could make a little error handler there and maybe exit, but it feels hacky. Alternatively, I could return an empty String which then errors out too, but it'll make a confusing error message. It'd prefer to see the stderr when it fails. 

So perhaps for this patch, make sure stderr is not discarded, but still fail with an exception?

Reply:
OK, I'll ACK it for now but would like to follow it up with a commit to either:

- fix `getFromNotmuchConfig` but "handle" the error (i.e. exception if error occurred) in `getDatabasePath` so that type does not change, OR
- absorb `getFromNotmuchConfig` into `getDatabasePath` and leave the behaviour as-is, because it's not used anywhere else.

Got a preference?

Reply:
Hm... how would you handle the error in `getFromNotmuchConfig` so that the type doesn't change? The only thing I can think of is to return an empty string in case of an error, which I don't think will help much?

Reply:
No, the type of `getFromNotmuchConfig` can change (so that it is total); but we move the non-totality (i.e. exception throwing) into `getDatabasePath`, so that *its* type does not change.

Reply:
Yep :+1:.  Or just keep using `readProcess_` but move it all into `getDatabasePath` because that's the only place it's called.

Reply:
Alright. Done.